### PR TITLE
250 integrate salt lake city cge

### PR DESCRIPTION
**Pull request by nunezco2**
Current status:

1. All the code for the Salt Lake City is in.
2. Current TODOs:
    a. Line 2195: finish generation of the output files. I looked at the Joplin example but it does not match very well how the final output is handled. Did not have the time to finish it up.
   b. Write the test_saltlakecge.py test.
   c. Test within pyincore (not as standalone, which works for now).
   d. Determine whether it is possible to pass a parameter to ipopt to increase multithreading.

Input and test output files follow in messages below.

### CODE REVIEW

**Comment by nunezco2** [comment]
Input files:

These need to be uploaded to the data service:
[Salt_Lake_FACTOR_2019.csv](https://github.com/IN-CORE/pyincore/files/10263475/Salt_Lake_FACTOR_2019.csv)
[Salt_Lake_HH_2019.csv](https://github.com/IN-CORE/pyincore/files/10263476/Salt_Lake_HH_2019.csv)
[Salt_Lake_JOBCORE_2019.csv](https://github.com/IN-CORE/pyincore/files/10263477/Salt_Lake_JOBCORE_2019.csv)
[Salt_Lake_OUTCR_2019.csv](https://github.com/IN-CORE/pyincore/files/10263478/Salt_Lake_OUTCR_2019.csv)
[Salt_Lake_CAPCOM_2019.csv](https://github.com/IN-CORE/pyincore/files/10263479/Salt_Lake_CAPCOM_2019.csv)
[Salt_Lake_SAM_2019.csv](https://github.com/IN-CORE/pyincore/files/10263480/Salt_Lake_SAM_2019.csv)
[sector_shocks.csv](https://github.com/IN-CORE/pyincore/files/10263481/sector_shocks.csv)

**Comment by nunezco2** [comment]
Output files:

These can be used for comparison.

 To run, make sure that `model_iterations = 1`

[domestic-supply.csv](https://github.com/IN-CORE/pyincore/files/10263485/domestic-supply.csv)
[gross-income.csv](https://github.com/IN-CORE/pyincore/files/10263486/gross-income.csv)
[household-count.csv](https://github.com/IN-CORE/pyincore/files/10263487/household-count.csv)
[post-disaster-factor-demand.csv](https://github.com/IN-CORE/pyincore/files/10263488/post-disaster-factor-demand.csv)
[pre-disaster-factor-demand.csv](https://github.com/IN-CORE/pyincore/files/10263489/pre-disaster-factor-demand.csv)

**Comment by nunezco2** [comment]
And, to create the test_*.py file, these are the current parameter-file assignments:

```
# SOCIAL ACCOUNTING MATRIX
SAM = pd.read_csv(os.path.join(filePath, 'Salt_Lake_SAM_2019.csv'), index_col=0)

# CAPITAL COMP
BB = pd.read_csv(os.path.join(filePath, 'Salt_Lake_CAPCOM_2019.csv'), index_col=0)

# MISC TABLES
miscPath = os.path.join(filePath, 'MiscFile')
JOBCR = pd.read_csv(os.path.join(miscPath, 'Salt_Lake_JOBCORE_2019.csv'), index_col=0)
MISCH = pd.read_csv(os.path.join(miscPath, 'Salt_Lake_HH_2019.csv'), index_col=0)
EMPLOY = pd.read_csv(os.path.join(miscPath, 'Salt_Lake_FACTOR_2019.csv'), index_col=0)
OUTCR = pd.read_csv(os.path.join(miscPath, 'Salt_Lake_OUTCR_2019.csv'), index_col=0)
```

File: tests/pyincore/analyses/saltlakecge/test_saltlakecge.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    saltlake_cge.set_parameter("model_iterations", 1)
+
+    saltlake_cge.set_input_dataset("SAM", sam)
+    saltlake_cge.set_input_dataset("BB", bb)
+    saltlake_cge.set_input_dataset("MISCH", misch)
+    saltlake_cge.set_input_dataset("EMPLOY", employ)
+    saltlake_cge.set_input_dataset("JOBCR", jobcr)
+    saltlake_cge.set_input_dataset("OUTCR", outcr)
+    saltlake_cge.set_input_dataset("sector_shocks", sector_shocks)
+    saltlake_cge.set_parameter("solver_path", "/home/cnavarro/git/Ipopt/src/Apps/AmplSolver/ipopt")
```
Comment (longshuicy) [inline tests/pyincore/analyses/saltlakecge/test_saltlakecge.py:45]: i think you can leave solver_path parameter out?

Comment (navarroc) [reply tests/pyincore/analyses/saltlakecge/test_saltlakecge.py]: Yes, my mistake there. It's fixed now.

**Review by longshuicy — APPROVED** [review]

**Review by jonglee1 — APPROVED** [review]

**Pull request merged**